File: sample-v7/src/main/java/io/erva/sample/base/cell/AlphaCell.java
package io.erva.sample.base.cell;

import android.view.View;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import io.erva.celladapter.Layout;
import io.erva.celladapter.v7.Cell;
import io.erva.sample.BaseCell;
import io.erva.sample.R;
import io.erva.sample.base.model.AlphaModel;

@Layout(R.layout.item_base_alpha)
public class AlphaCell extends BaseCell<AlphaModel, AlphaCell.Listener> {

    @BindView(R.id.tv_alpha) TextView textView;

    public AlphaCell(View view) {
        super(view);
    }

    @Override
    protected void bindView() {
        textView.setText(getItem().getAlpha());
    }

    @OnClick(R.id.btn_one_press)
    public void onPressOne() {
        if (getListener() != null) getListener().onPressOne(getItem());
    }

    @OnClick(R.id.btn_two_press)
    public void onPressTwo() {
        if (getListener() != null) getListener().onPressTwo(getItem());
    }

    public interface Listener extends Cell.Listener<AlphaModel> {

        void onPressOne(AlphaModel model);

        void onPressTwo(AlphaModel model);
    }
}

File: sample-kotlin/src/main/kotlin/io/erva/sample/base/cell/AlphaCell.kt
package io.erva.sample.base.cell

import android.view.View
import io.erva.celladapter.Cell
import io.erva.celladapter.Layout
import io.erva.sample.R
import io.erva.sample.base.model.AlphaModel
import kotlinx.android.synthetic.main.item_base_alpha.view.*

@Layout(R.layout.item_base_alpha)
class AlphaCell(view: View) : Cell<AlphaModel, AlphaCell.Listener>(view) {

    override fun bindView() {
        view.tv_alpha.text = item().alpha
        view.btn_one_press.setOnClickListener { listener()?.onPressOne(item()) }
        view.btn_two_press.setOnClickListener { listener()?.onPressTwo(item()) }
    }

    interface Listener : Cell.Listener<AlphaModel> {

        fun onPressOne(item: AlphaModel)

        fun onPressTwo(item: AlphaModel)
    }
}